great_expectations/rule_based_profiler/config/base.py

```diff
@@ -688,10 +688,9 @@ class Meta:
         required=True,
         allow_none=False,
     )
-    id_ = fields.String(
+    id = fields.String(
         required=False,
-        allow_none=False,
-        data_key="id",
+        allow_none=True,
```

Comment:
I believe this was erroneously set to allow_none=False, which is needed to be backward compatible for existing id-less configs.